src/tools/singlejar/output_jar.cc

```diff
@@ -119,12 +120,12 @@ int OutputJar::Doit(Options *options) {
     // TODO(asmundak):  Consider going back to sendfile() or reflink
     // (BTRFS_IOC_CLONE/XFS_IOC_CLONE) here.  The launcher preamble can
     // be very large for targets with many native deps.
-    ssize_t byte_count = AppendFile(in_fd, 0, statbuf.st_size);
+    ptrdiff_t byte_count = AppendFile(in_fd, 0, statbuf.st_size);
```

Comment:
_Comment from internal reviewer:_

AppendFile is still declared as returning ssize_t, how is this working in MSVC?
IMHO a better approach is to define ssize_t when compiling with MSVC.

Reply:
I have a bigger patch locally I haven't upstream to convert all `ssize_t` and `off_t` to C standard type `ptrdiff_t`.

`ssize_t` is a POSIX type that MSVC does not have. MSVC x64 does have `off_t`, but define as `int32_t` instead of `int64_t` due to legacy reason, but that will prevent us from handling large file. `ptrdiff_t` is more consistent.

For 64 bit Unix system, `ssize_t`, `off_t` and `ptrdiff_t` are all typedef of `long long/int64_t`, so this shouldn't break any Unix code.

Reply:
_Forwarded._

Reply:
I will do the change in the larger PR. Done.

Reply:
_Forwarded._